### PR TITLE
Added Layout.CreateFromMethod to allow custom layoutrenders using Config API

### PR DESCRIPTION
**Pull request by snakefoot**
Another building block to make it easier to use the Config API

- Added Layout.CreateFromString and marked Layout.FromString as obsolete
- Added Format-property to FuncLayoutRenderer 
- Added IStringValueRenderer-support to FuncLayoutRenderer

### CODE REVIEW

**Comment by 304NotModified** [comment]
cool!

To fully understand, what is the use case for `Layout.FromLayoutMethod`? (as and end-user)?

**Comment by snakefoot** [comment]
Lets imagine https://github.com/nlog/nlog/wiki/JsonLayout#from-the-api

```c#
var jsonLayout = new JsonLayout
{
    Attributes =
    {
        new JsonAttribute("myFunkyValue", Layut.FromLayoutMethod(logEvent => 42),
    }
}

```

**Comment by 304NotModified** [comment]
Proposal, rename to CreateFromMethod

**Comment by 304NotModified** [comment]
`.Create` would be also fine for me

File: src/NLog/Layouts/Layout.cs [hunk 1]
```diff
@@ -114,6 +114,28 @@ public static Layout FromString(string layoutText, ConfigurationItemFactory conf
             return new SimpleLayout(layoutText, configurationItemFactory);
         }
 
+        /// <summary>
+        /// Implicits converts the specific lambda method into a <see cref="SimpleLayout"/>.
+        /// </summary>
+        /// <param name="layoutMethod">Method that renders the layout.</param>
+        /// <param name="threadSafe">Tell if method is safe for concurrent threading.</param>
+        /// <returns>Instance of <see cref="SimpleLayout"/>.</returns>
+        public static Layout FromLayoutMethod(Func<LogEventInfo, object> layoutMethod, bool threadSafe = false)
```
Comment (304NotModified) [inline src/NLog/Layouts/Layout.cs]: Consider replacing "threadSafe " with an object (e.g. LayoutCreationOptions). That would be easier to extend in the future)

Comment (snakefoot) [reply src/NLog/Layouts/Layout.cs]: Have introduced parameter LayoutRenderOptions instead of `bool threadSafe`

**Comment by 304NotModified** [comment]
Ow oops. Missed the fromString. Do think create.. is better? (I think so, but obsoletes and/or inconsistent apis aren't preferred!)

**Comment by snakefoot** [comment]
Think Create is nice. But consistency is more important. Since I want to make ConfigurationItemFactory internal in NLog 5.0 then I want to obsolete one of Layout.FromString methods anyway.


Sent from my Sony Xperia


---- Julian Verdurmen wrote ----


Ow oops. Missed the fromString. Do think create.. is better? (I think so, but obsoletes and/or inconsistent apis aren't preferred!)

**Comment by 304NotModified** [comment]
OK merged :)

**Pull request merged**